src/Options/Auto.hpp

```diff
@@ -92,27 +93,52 @@ std::ostream& operator<<(std::ostream& os, const Auto<T, Label>& x) {
   }
 }
 
+namespace Auto_detail {
+template <typename Label>
+struct AutoLabel {};
+}  // namespace Auto_detail
+
+template <typename Label>
+struct create_from_yaml<Auto_detail::AutoLabel<Label>> {
+  template <typename Metavariables>
+  static Auto_detail::AutoLabel<Label> create(const Option& options) {
+    const auto label_string = pretty_type::name<Label>();
+    try {
+      if (options.parse_as<std::string>() == label_string) {
+        return {};
+      }
+    } catch (...) {
+      // The node failed to parse as a string.  It is not the Label.
+    }
+    // The error if the std::variant parse fails will print the value
+    // from the input file (and the T parse probably will too), so we
+    // don't need to print it again.
+    PARSE_ERROR(options.context(),
+                "Failed to parse as Auto label \"" << label_string << "\"");
+  }
+};
+
 template <typename T, typename Label>
 struct create_from_yaml<Auto<T, Label>> {
   template <typename Metavariables>
   static Auto<T, Label> create(const Option& options) {
-    try {
-      if (options.parse_as<std::string>() == pretty_type::name<Label>()) {
+    auto parsed_variant =
+        options.parse_as<std::variant<Auto_detail::AutoLabel<Label>, T>,
+                         Metavariables>();
+    if (std::holds_alternative<T>(parsed_variant)) {
+      return Auto<T, Label>{std::move(std::get<T>(parsed_variant))};
```

Comment:
This won't work for `Options::Auto<std::string>`. Not sure if that's relevant or not.

Reply:
Why not?  It seems to work.  I added a test.

Reply:
Oh good, yea I missed that the variant still has two distinct types (I thought it would be string twice).